File: docs/snippets/cpp/2576_find_the_maximum_number_of_marked_indices.cpp
#include <vector>
#include <algorithm>
#include <cstdio>

using namespace std;

int maxNumOfMarkedIndices(vector<int> &nums)
{
    sort(nums.begin(), nums.end());
    int n = nums.size();
    int slow = 0;
    int fast = n / 2;
    int count = 0;

    while (slow < n / 2 && fast < n)
    {
        if (nums[fast] >= 2 * nums[slow])
        {
            count += 2;
            slow++;
        }
        fast++;
    }
    return count;
}

int main()
{
    vector<int> nums = {1, 3, 2, 3, 1};
    int result = maxNumOfMarkedIndices(nums);
    printf("Result: %d\n", result);
    return 0;
}

File: src/2576_find_the_maximum_number_of_marked_indices.py
from typing import List


# Fast Slow Pointers
def maxNumOfMarkedIndices(nums: List[int]) -> int:
    nums.sort()
    n = len(nums)
    slow, fast = 0, n // 2
    count = 0

    while slow < n // 2 and fast < n:
        if nums[fast] >= 2 * nums[slow]:
            count += 2
            slow += 1
        fast += 1

    return count


nums = [3, 5, 2, 4]
print(maxNumOfMarkedIndices(nums))  # 2
